fbchat/get.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: UTF-8 -*-
+
+from __future__ import unicode_literals
+from .base import Base
+
+
+class Get(Base):
+    """Enables retrieving information about threads and messages"""
+
+    def get_threads(self, limit=None):
+        """Retrieve the threads that the client is currently chatting with
+
+        If ``limit`` is ``None``, the result will be a generator, otherwise the
+        result will be a list with maximum length of ``limit``
+
+        Args:
+            limit (int): Max. number of threads to retrieve
+
+        Return:
+            List of `Thread` objects or a generator yielding `Thread` objects
+        """
+
+    def get_archived_threads(self, limit=None):
+        """Retrieve the client's archived threads
+
+        ``limit`` and return values behave as in `get_messages`
+        """
+
+    def get_filtered_threads(self, limit=None):
+        """Retrieve the client's filtered threads
+
+        ``limit`` and return values behave as in `get_messages`
+        """
+
+    def get_unread_threads(self, limit=None):
+        """Retrieve the client's unread threads
+
+        ``limit`` and return values behave as in `get_messages`
+        """
+
+    def get_friends(self, limit=None):
+        """Retrieve the users that the client is friends with
+
+        ``limit`` and return values behave as in `get_messages`
+        """
+
+
+    def get_threads_from_ids(self, *thread_ids):
+        """Retrieve threads based on their IDs
+
+        If ``thread_ids`` contains a single iterable, then a generator yielding
+        the threads is returned.
+
+        If ``thread_ids`` is a single value or multiple values, then a list
+        containing the threads is returned.
+
+        Args:
+            *thread_ids: One or more thread IDs to query
+
+        Return:
+            `Thread` objects, in the order and format their IDs were supplied
+        """
+
+    def get_messages_from_ids(self, *message_ids):
+        """Retrieve messages based on their IDs
+
+        If ``message_ids`` contains a single iterable, then a generator
+        yielding the messages is returned.
+
+        If ``message_ids`` is a single value or multiple values, then a list
+        containing the messages is returned.
+
+        Args:
+            *message_ids: One or more message IDs to query
+
+        Return:
+            `Message` objects, in the order and format their IDs were supplied
+        """
+
+
+    def get_messages(self, thread, limit=None):
```

Comment:
So, we can retrieve messages, but what about events? By event I mean the small pieces of "history", that eg. says "X left the group". The actions that `ThreadControl` and `ThreadInterraction` can do and see, as they're happening, but we have no way of looking back, and seeing when they happened.